### ISSUE TITLE
Update statement: An item with the same key has already been added

### ISSUE THREAD

**Issue by FH-Inway** [MarkMpn/Sql4Cds#138]
On one of my environments, when I try to run an update statement, I get the error message "An item with the same key has already been added."

For all my other environments, the update statement works.

![image](https://user-images.githubusercontent.com/33372796/145026151-1acea87a-95a5-4db3-b73a-00e31ffdc73d.png)

Entry in MarkMpn.Sql4Cds.log:

```
2021-12-07 01:01:25.846 		Error	System.ArgumentException: Ein Element mit dem gleichen Schlüssel wurde bereits hinzugefügt.
   bei System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   bei System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   bei System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   bei MarkMpn.Sql4Cds.Engine.ExecutionPlanBuilder.ConvertUpdateStatement(UpdateSpecification update, IList`1 hints)
   bei MarkMpn.Sql4Cds.Engine.ExecutionPlanBuilder.Build(String sql)
   bei MarkMpn.Sql4Cds.SqlQueryControl.backgroundWorker_DoWork(Object sender, DoWorkEventArgs e)
   bei System.ComponentModel.BackgroundWorker.OnDoWork(DoWorkEventArgs e)
   bei System.ComponentModel.BackgroundWorker.WorkerThreadStart(Object argument)
```

XrmToolBox version is 1.2021.9.52
SQL 4 CDS version is 5.3.1.0

**Comment by MarkMpn** [comment]
It sounds like the view of the metadata in this environment has got messed up. Can you try right-clicking on the environment name in the Object Explorer pane and clicking Refresh please?

**Comment by FH-Inway** [comment]
Awesome answer time :+1:

I tried the refresh, but the issue persists. When I did the refresh the first time, it showed a small yellow dialog telling me that the refresh is in progress. It took about a minute. Now when I do the refresh, the dialog only appears for a short moment.

I did the first refresh before the issue occured, because I had imported a solution in the environment that added a new table field on account.
